operator/src/main/resources/scripts/utils.sh

```diff
@@ -419,6 +419,147 @@ function getAdminServerUrl() {
   echo ${admin_protocol}://${AS_SERVICE_NAME}:${ADMIN_PORT}
 }
 
+function waitForShutdownMarker() {
+  #
+  # Wait forever.   Kubernetes will monitor this pod via liveness and readyness probes.
+  #
+  trace "Wait indefinitely so that the Kubernetes pod does not exit and try to restart"
+  while true; do
+    if [ -e ${SHUTDOWN_MARKER_FILE} ] ; then
+      exit 0
+    fi
+    sleep 3
+  done
+}
+
+#
+# Define helper fn for failure debugging
+#   If the livenessProbeSuccessOverride file is available, do not exit from startServer.sh.
+#   This will cause the pod to stay up instead of restart.
+#   (The liveness probe checks the same file.)
+#
+
+function exitOrLoop {
+  if [ -f /weblogic-operator/debug/livenessProbeSuccessOverride ]
+  then
+    waitForShutdownMarker
+  else
+    exit 1
+  fi
+}
+
+#
+# Define helper fn to create a folder
+#
+
+function createFolder {
+  mkdir -m 750 -p $1
+  if [ ! -d $1 ]; then
+    trace SEVERE "Unable to create folder $1"
+    exitOrLoop
+  fi
+}
+
+# Returns the count of the number of files in the specified directory
+function countFilesInDir() {
+  set -o pipefail
```

Comment:
If you intend to scope pipefail just to this method, then I think you may need to put a new line with an open paren at the beginning of the function and a closing paren at the end.

Reply:
If you intend to scope pipefail to all scripts (which I think would be just find), then move the set -o pipefail to the top of utils.sh.

Reply:
moved 'set -o pipefail' to the top of utils.sh.